src/zope/file/ftesting.zcml

```diff
@@ -1,72 +1,134 @@
+  <include package="zope.app.basicskin" />
+
+  <include package="zope.app.publication" />
+  <include package="zope.app.pagetemplate" />
+  <include package="zope.password" />
+  <include package="zope.browserresource" />
+  <include package="zope.browsermenu" />
+
+
+  <include package="zope.securitypolicy" file="meta.zcml" />
+  <include package="zope.securitypolicy" />
+  <include package="zope.login" />
+
+
+  <browser:menu
+      id="zmi_views"
+      title="Views"
+      description="Menu for displaying alternate representations of an object"
+      />
+
+  <include package="zope.file" file="browser.zcml"/>
+  <include package="zope.file" file="menus.zcml"/>
+  <include package="zope.file"/>
+
+  <include package="zope.formlib"/>
+  <include package="zope.mimetype" file="meta.zcml"/>
+  <include package="zope.mimetype"/>
+
+  <!-- Needs zope.app.localpermission to be ported -->
+  <!-- <include package="zope.app.security" />  -->
+  <browser:defaultView name="index.html" />
+
+  <!-- This normally comes from zope.app.folder -->
+  <browser:menu
+      id="zmi_actions"
+      title="Actions"
+      description="Menu for displaying actions to be performed"
+      />
+  <browser:addMenuItem
+      class="zope.site.folder.Folder"
+      title="Folder"
+      description="Minimal folder"
+      permission="zope.ManageContent"
+      />
+
+  <browser:view
+      name="+"
+      menu="zmi_actions" title="Add"
+      for="zope.site.interfaces.IFolder"
+      permission="zope.ManageContent"
+      class=".testing.Adding"
+      />
+  <browser:page
+      for="zope.site.interfaces.IFolder"
+      name="contents.html"
+      class=".testing.Contents"
+      permission="zope.ManageContent"
+      attribute="contents"
+      />
+
+  <browser:page
+      for="*"
+      name="SelectedManagementView.html"
+      permission="zope.Public"
+      class=".testing.ManagementViewSelector"
+      allowed_interface="zope.publisher.interfaces.browser.IBrowserPublisher"
+      />
+
+  <securityPolicy
+      component="zope.securitypolicy.zopepolicy.ZopeSecurityPolicy" />
+
+  <role id="zope.Anonymous" title="Everybody"
+        description="All users have this role implicitly" />
+  <role id="zope.Manager" title="Site Manager" />
+
+  <!-- Replace the following directive if you don't want public access -->
+  <grant permission="zope.View"
+         role="zope.Anonymous" />
+  <grant permission="zope.dublincore.view"
+         role="zope.Anonymous" />
+
+  <grantAll role="zope.Manager" />
+
+  <!-- Principals -->
+
+  <unauthenticatedPrincipal
+      id="zope.anybody"
+      title="Unauthenticated User" />
+
+  <!-- Principal that tests generally run as -->
+  <principal
+      id="zope.mgr"
+      title="Manager"
+      login="mgr"
+      password="mgrpw" />
+  <grant role="zope.Manager" principal="zope.mgr" />
+
+  <class class="zope.site.folder.Folder">
+	<allow attributes="items __getitem__ get __contains__ __setitem__" />
```

Comment:
2-space indents for child elements (as opposed to 4-space indents for attributes) in ZCML.

This looks like a single tab character to me?  (GitHub doesn't show tabs, but when I'm selecting it with the mouse I see just one wide selectable whitespace thing.)